bin/reth/src/commands/db/checksum.rs

```diff
@@ -41,10 +54,30 @@ impl<DB: Database> ChecksumViewer<'_, DB> {
         let tx = provider.tx_ref();
 
         let mut cursor = tx.cursor_read::<RawTable<T>>()?;
-        let walker = cursor.walk(None)?;
+        let walker = match (self.start_key.as_deref(), self.end_key.as_deref()) {
+            (Some(start), Some(end)) => {
+                info!("start={start} \n end={end}");
+                let start_key = table_key::<T>(start).map(RawKey::<T::Key>::new)?;
+                let end_key = table_key::<T>(end).map(RawKey::<T::Key>::new)?;
+                cursor.walk_range(start_key..=end_key)?
+            }
+            (None, Some(end)) => {
+                info!("start=.. \n end={end}");
+                let end_key = table_key::<T>(end).map(RawKey::<T::Key>::new)?;
+
+                cursor.walk_range(..=end_key)?
+            }
+            (Some(start), None) => {
+                info!("start={start} \n end= ");
+                let start_key = table_key::<T>(start).map(RawKey::<T::Key>::new)?;
+                cursor.walk_range(start_key..)?
+            }
+            (None, None) => cursor.walk_range(..)?,
+        };
 
         let start_time = Instant::now();
-        let mut hasher = AHasher::default();
+        let mut hasher = RandomState::with_seeds(1, 2, 3, 4).build_hasher();
```

Comment:
AHasher::default still will rely a random state which is generated either compile time or runtime, so that everytime we compile the client and rerun with the same command, the checksum changed, which is not what we want IMO.

Reply:
> AHasher::default still will rely a random state which is generated either compile time or runtime, so that everytime we compile the client and rerun with the same command, the checksum changed, which is not what we want IMO.

ah right, I think we can disable this behavior if we make sure it does not use certain features, but this should be better.